nested-types.ipynb

```diff
@@ -0,0 +1,915 @@
+       "- length: 4\n",
+       "- offset: 0\n",
+       "- null_count: 1\n",
+       "- buffers[2]:\n",
+       "  - validity <bool[1 b] 11010000>\n",
+       "  - data_offset <int32[20 b] 0 1 4 4 6>\n",
+       "- dictionary: NULL\n",
+       "- children[1]:\n",
+       "  - <nanoarrow.c_lib.CArrayView>\n",
+       "    - storage_type: 'struct'\n",
+       "    - length: 6\n",
+       "    - offset: 0\n",
+       "    - null_count: 0\n",
+       "    - buffers[1]:\n",
+       "      - validity <bool[0 b] >\n",
+       "    - dictionary: NULL\n",
+       "    - children[2]:\n",
+       "      - <nanoarrow.c_lib.CArrayView>\n",
+       "        - storage_type: 'string'\n",
+       "        - length: 6\n",
+       "        - offset: 0\n",
+       "        - null_count: 0\n",
+       "        - buffers[3]:\n",
+       "          - validity <bool[0 b] >\n",
+       "          - data_offset <int32[28 b] 0 11 22 38 46 62 70>\n",
+       "          - data <string[70 b] b'D...>\n",
+       "        - dictionary: NULL\n",
+       "        - children[0]:\n",
+       "      - <nanoarrow.c_lib.CArrayView>\n",
+       "        - storage_type: 'int32'\n",
+       "        - length: 6\n",
+       "        - offset: 0\n",
+       "        - null_count: 1\n",
+       "        - buffers[2]:\n",
+       "          - validity <bool[1 b] 11111000>\n",
+       "          - data <int32[24 b] 10 8 4 5 10 0>\n",
+       "        - dictionary: NULL\n",
+       "        - children[0]:"
+      ]
+     },
+     "execution_count": 36,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "na.c_array_view(na.c_array(column_6))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Comparing the conversion of struct and map to python objects\n",
+    "\n",
+    "Struct converts to a list of dictionaries:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([{'x': 'joe', 'y': 1.0}, {'x': None, 'y': 2.0}, None,\n",
+       "       {'x': 'mark', 'y': 4.0}, {'x': 'jane', 'y': None}], dtype=object)"
+      ]
+     },
+     "execution_count": 46,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "column_4.to_numpy(zero_copy_only=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[{'x': 'joe', 'y': 1},\n",
+       " {'x': None, 'y': 2},\n",
+       " None,\n",
+       " {'x': 'mark', 'y': 4},\n",
+       " {'x': 'jane', 'y': None}]"
+      ]
+     },
+     "execution_count": 49,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "column_4.to_pylist()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And map converts to a nested list of tuples:"
```

Comment:
```suggestion
    "And map converts to a nested list of tuples by default:"
```

And then I would also show the example of `map_as_pydict` keyword that you can get back the dictionaries as well

Reply:
Note: for this example, pandas needs to be added to the setup instructions.